community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/ExecutionEngine.scala

```diff
@@ -216,6 +218,17 @@ class ExecutionEngine(val queryService: GraphDatabaseQueryService,
     throw new IllegalStateException("Could not execute query due to insanely frequent schema changes")
   }
 
+  @throws(classOf[ParameterNotFoundException])
+  private def checkParameters(queryParams: Seq[String], givenParams: MapValue, extractedParams: Map[String, Any]) {
+    exceptionHandler.runSafely {
+      queryParams.foreach( key => {
+        if (!givenParams.containsKey(key) && !extractedParams.contains(key)) {
+          throw new ParameterNotFoundException("Expected a parameter named " + key)
```

Comment:
Would it not be even more useful to collect all missing parameters and throw an exception listing them all, instead of throwing one naming only the first missing parameter?

Reply:
That is a good idea, will fix it.